File: PROJECT/WEB/CODE/src/lib/service/user_service.js
// -- IMPORTS

import { logError } from 'senselogic-gist';
import { database } from '$lib/database';

// -- TYPES

class UserService
{
    // -- OPERATIONS

    async signUpUser(
        email,
        password
        )
    {
        const { user, error }
            = await database.auth.signUp(
                  {
                      email,
                      password
                  }
                  );

        if ( error !== null )
        {
            logError( error );
        }

        return user;
    }

    // ~~

    async signInUser(
        email,
        password
        )
    {
        const { user, error }
            = await database.auth.signIn(
                  {
                      email,
                      password
                  }
                  );

        if ( error !== null )
        {
            logError( error );
        }

        return user;
    }

    // ~~

    async signOutUser(
        )
    {
        const { error }
            = await database.auth.signOut();

        if ( error !== null )
        {
            logError( error );
        }
    }
}

// -- VARIABLES

export const userService
    = new UserService();

File: PROJECT/MOBILE/CODE/lib/service/user_service.dart
// -- IMPORTS

import 'package:supabase_flutter/supabase_flutter.dart';
import '../database/database.dart';

// -- TYPES

class UserService
{
    // -- INQUIRIES

    Future<bool> signIn(
        String email,
        String password
        ) async
    {
        final response = await database.auth.signInWithPassword( email: email, password: password );

        session = response.session;
        user = response.user;

        return ( user != null );
    }

    // ~~

    Future<bool> signUp(
        String email,
        String password
        ) async
    {
        final response = await database.auth.signUp( email: email, password: password );

        session = response.session;
        user = response.user;

        return ( user != null );
    }

    // ~~

    Future<bool> signOut(
        ) async
    {
        await database.auth.signOut();

        user = null;

        return true;
    }
}

// -- VARIABLES

final
    userService = UserService();
late User?
    user;
late Session?
    session;
